app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/incline_direction/AddStepsIncline.kt

```diff
@@ -25,6 +30,14 @@ class AddStepsIncline : OsmFilterQuestType<Incline>(), AndroidQuest {
 
     override fun getTitle(tags: Map<String, String>) = R.string.quest_steps_incline_title
 
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> = mapData
+        .filter(filter)
+        .filter { it is Way && !it.isClosed }
+        .asIterable()
+
+    override fun isApplicableTo(element: Element) =
+        if (!filter.matches(element)) false else null
```

Comment:
why `null`? This seems incorrect.

Also, you could just

```suggestion
    override fun getApplicableElements(mapData: MapDataWithGeometry) = 
        mapData.ways.filter { isApplicableTo(it) }

    override fun isApplicableTo(element: Element) =
        filter.matches(element) && !element.isArea()
```